Wikipedia/Code/ArticleViewController.swift

```diff
@@ -123,7 +121,6 @@ class ArticleViewController: ViewController, HintPresenting {
     lazy var webViewConfiguration: WKWebViewConfiguration = {
         let configuration = WKWebViewConfiguration()
         configuration.processPool = ArticleViewController.webProcessPool
-        configuration.setURLSchemeHandler(schemeHandler, forURLScheme: schemeHandler.scheme)
```

Comment:
Our article view is heavily dependent on this scheme handler for supporting offline mode in saved articles. Without this, visiting a saved article without a network connection no longer displays. Rather than touching this area, I suggest trying to modify the links at the moment they are copied. Here are some approaches that might work:

1. Suppress the contextual menu's default "Copy" button (in [WMFWebView.m](https://github.com/wikimedia/wikipedia-ios/blob/160c00fa4e4e104854e4057697c57480a504244a/Wikipedia/Code/WMFWebView.m#L6), we're already doing this with `_share`) and provide our own custom one with implementation (in `ArticleViewController+ContextMenu.swift`, [updateMenuItems](https://github.com/wikimedia/wikipedia-ios/blob/160c00fa4e4e104854e4057697c57480a504244a/Wikipedia/Code/ArticleViewController%2BContextMenu.swift#L2) method. Again see `shareMenuItem` here as an example.). In this implementation we'll probably need to go across the javascript bridge to copy the selected text in the web view, then replace any instances of "app://" with "https://", then save it to the [clipboard](https://developer.apple.com/documentation/uikit/uipasteboard).
2. Override `WMFWebView`'s or `ArticleViewController`'s `copy:(id)sender` [method](https://developer.apple.com/documentation/uikit/uiresponderstandardeditactions/2354191-copy) and do some custom handling to tweak the links. This seems to get called right when the copy button is pressed.
3. Listen for `UIPasteboard`'s `changedNotification` (doc [here](https://developer.apple.com/documentation/uikit/uipasteboard/1622104-changednotification)) in `ArticleViewController` and modify the [clipboard](https://developer.apple.com/documentation/uikit/uipasteboard) in your handler method.

I tried each of these approaches and got closest with the 3rd, so that might be the one to aim for.